db/trie/merklepatriciatree/extensionnode.go

```diff
@@ -135,25 +138,35 @@ func (e *extensionNode) commonPrefixLength(key []byte) uint8 {
 	return commonPrefixLength(e.path, key)
 }
 
-func (e *extensionNode) updatePath(path []byte) (*extensionNode, error) {
+func (e *extensionNode) updatePath(path []byte, h bool) (node, error) {
```

Comment:
h means isLeaf?

Reply:
i think h is a flag whether the return value is hashnode or extension node itself